lib/libutee/tee_api_property.c

```diff
@@ -387,27 +426,42 @@ TEE_Result TEE_GetPropertyAsBinaryBlock(TEE_PropSetHandle propsetOrEnumerator,
 	struct prop_value pv;
 	void *val;
 	int val_len;
-	size_t size;
+	uint32_t size;
 
-	if (valueBuffer == NULL || valueBufferLen == NULL)
-		return TEE_ERROR_BAD_PARAMETERS;
+	if (valueBuffer == NULL || valueBufferLen == NULL) {
+		res = TEE_ERROR_BAD_PARAMETERS;
+		goto err;
+	}
 
 	res = propget_get_property(propsetOrEnumerator, name, &pv);
 	if (res != TEE_SUCCESS)
-		return res;
+		goto err;
 
-	if (pv.type != USER_TA_PROP_TYPE_BINARY_BLOCK)
-		return TEE_ERROR_BAD_FORMAT;
+	if (pv.type != USER_TA_PROP_TYPE_BINARY_BLOCK) {
+		res = TEE_ERROR_BAD_FORMAT;
+		goto err;
+	}
 
 	val = pv.u.str_val;
 	val_len = strlen(val);
 	size = *valueBufferLen;
-	if (!base64_dec(val, val_len, valueBuffer, &size))
+	if (!base64_dec(val, val_len, valueBuffer, &size)) {
```

Comment:
base64_dec() expects `size_t *` in 4th parameter, now you're supplying `uint32_t *` instead.